selfie.c

```diff
@@ -773,6 +773,127 @@ void initSyscalls() {
     SYSCALL_GETCHAR = 5002;
 }
 
+
+
+// *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~
+// -----------------------------------------------------------------
+// ---------------------------     O S   ---------------------------
+// -----------------------------------------------------------------
+// *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~
+
+int  pid;
+int *head;
+int *tail;
+
+void initialize_process_queue() {
```

Comment:
OK for now, but: you should separate list logic from node semantic, i.e., the list should not bother it's nodes (what they are, what they mean), and the nodes should not bother which list they're in. Think of the next step: You've got another list of processes, say, the blocked queue. How would you move a node from one queue to another?